File: backend/Models/InfoBook.cs
using System.ComponentModel.DataAnnotations;

namespace lab8.Models
{
    public class InfoBook
    {
        [Key]
        public int ID { get; set; }
        public int BookID { get; set; }
        public int TicketID { get; set; }
        public DateTime DateTakeBook { get; set; }
        public DateTime? DateReturnBook { get; set; }
        public Ticket Ticket { get; set; } = null!;
        public Book Book { get; set; } = null!;
    }
}

File: frontend/app/models/infobook.ts
import { bindCallback } from "rxjs";
import { Book } from "./book";
import { Ticket } from "./ticket";

export class InfoBook{
    ID!: number;
    BookID!: number;
    TicketID!: number;
    DateTakeBook!: Date;
    DateReturnBook!: Date;
    Book!: Book;
    Ticket!: Ticket;

    constructor(id: number, bookID: number, ticketID: number, dateTakeBook: Date, 
        dateReturnBook: Date, book:Book, ticket:Ticket ){

            this.ID = id;
            this.BookID = bookID;
            this.TicketID = ticketID;
            this.DateTakeBook = dateTakeBook;
            this.DateReturnBook = dateReturnBook;
            this.Book = book;
            this.Ticket = ticket;
        }
}